src/source/CxMarshal.scala

```diff
@@ -302,10 +302,11 @@ class CxMarshal(spec: Spec) extends Marshal(spec) {
           case MOptional => throw new AssertionError("nested optional?")
           case p: MPrimitive => (p.cxBoxed, true)
           //   case MString => ("Platform::IBox<Platform::String^>", true)
-          case e: MExtern =>
-            (s"Platform::IBox<${expr(arg, namespace, e.cx.reference)._1}>", true)
-          case m =>
-            (s"Platform::IBox<${expr(arg, namespace, true)._1}>", true)
+          case d: MDef if DEnum == d.defType =>
```

Comment:
nope, DEnum isn't an arg.base type, it's a definition type on a MDef... or something, either way it doesn't compile if i try to just case d: DEnum